torch/fx/proxy.py

```diff
@@ -259,18 +259,32 @@ def __len__(self):
                            "this call to be recorded, please call torch.fx.wrap('len') at "
                            "module scope")
 
-    def __torch_function__(self, orig_method, types, args=None, kwargs=None):
+    @classmethod
+    def __torch_function__(cls, orig_method, types, args=None, kwargs=None):
         args = args if args else ()
         kwargs = kwargs if kwargs else {}
 
+        tracers : Dict[Any, None] = {}
+
+        def find_tracer(a):
+            if isinstance(a, cls):
+                tracers[a.tracer] = None
+        torch.fx.node.map_aggregate(args, find_tracer)
+        torch.fx.node.map_aggregate(kwargs, find_tracer)
+
+        if len(tracers) > 1:
+            raise RuntimeError(f'Found multiple different tracers {list(tracers.keys())} while '
+                               'trying to trace operations {orig_method}')
```

Comment:
I had to test this behavior because i didn't know what would happen but I think the continuation of the fstring needs the `f` sigil on it as well:

```
a = 3
b = 4

fstring = (f'i am a {a} '
            'fstring {b}')

print(fstring)
"""
i am a 3 fstring {b}
"""
```

Reply:
Ho that's interesting. Let update that.